aggregate_root/lib/aggregate_root.rb

```diff
@@ -47,7 +47,12 @@ def apply(*events)
       end
     end
 
+    def adding_untested_code_to_tigger_failure
+      # yeah
+    end
+
     def version
+      adding_untested_code_to_tigger_failure
```

Comment:
I'm still unfamiliar with your project, normally this gets picked up and causes a build failure.

Reply:
monorepo excercises edge cases 😉

```ruby
      def touches?(path, line_range)
        pp path
        pp touched_paths.keys
        pp touched_paths.fetch(path)

        touched_paths
          .fetch(path) { return false }
          .touches?(line_range)
      end
```

```
bundle exec mutant run --since master -raggregate_root 'AggregateRoot::AggregateMethods#version'

# path
#<Pathname:/Users/pawel/Code/rails_event_store/aggregate_root/lib/aggregate_root.rb> 

# touched_paths
[#<Pathname:/Users/pawel/Code/rails_event_store/aggregate_root/.circleci/config.rb>,
 #<Pathname:/Users/pawel/Code/rails_event_store/aggregate_root/.circleci/config.yml>,
 #<Pathname:/Users/pawel/Code/rails_event_store/aggregate_root/aggregate_root/lib/aggregate_root.rb>,
 #<Pathname:/Users/pawel/Code/rails_event_store/aggregate_root/support/make/mutant.mk>]

# fetch
KeyError: key not found: #<Pathname:/Users/pawel/Code/rails_event_store/aggregate_root/...
```

```
$ git diff-index master                                         1s 208ms 
:100644 100644 accbf9736bd717d35310eecbae58a2cfba302182 a99c53200245d31c5d4b3eebf00947feb4372b10 M	.circleci/config.rb
:100644 100644 d587ddbbd8b49cf32dd28ac9073aad9cfd2df51e cb115e917fb6e2741f87998521ef508e061422be M	.circleci/config.yml
:100644 100644 4e327bd60388a7fffd3036d617d03962a08b045d 7171cc46917857edc9de78ed8a12b31cd4f2029f M	aggregate_root/lib/aggregate_root.rb
:100644 100644 d60c244d6445a40cf226f8b8696e2af048139b6e 46855a33f84b57413429a568db4aa557ce250bae M	support/make/mutant.mk
```

This one is odd:
/Users/pawel/Code/rails_event_store/**aggregate_root**/aggregate_root/lib/aggregate_root.rb

Reply:
Actually all `touched_paths` are odd and incorrectly nested under `aggregate_root`

Reply:
All running from:
```
$ pwd                                                                    
/Users/pawel/Code/rails_event_store/aggregate_root
```

Reply:
@mbj 
so `git diff-index` shows paths relative to project root, mutant interprets those entries as if the root was subdirectory from which `mutant run` is executed

Reply:
Thats easy to improve to make mutant aware of non project root invokations.

Reply:
@mbj proposed https://github.com/mbj/mutant/pull/1139